controller.py

```diff
@@ -243,15 +245,257 @@ def get_system_ram():
     return mem
 
 
+def create_backup(args, queue_output, config_qemu, raw_faultlist, expanded_faultlist):
+    """
+    Creates a backup dict with hashes and pushes it to the given queue
+
+    :param args: parameters of the program
+    :param queue_output: queue to push the created backup
+    :param config_qemu: qemu.json config dictionary
+    :param raw_faultlist: list of raw faults from the json file
+    :param expanded_faultlist: list of expanded faults
+    :return: void
+    """
+    filenames = [
+        args.qemu.name,
+        args.faults.name,
+        config_qemu["kernel"],
+        config_qemu["bios"],
+    ]
+
+    hasher = hashlib.sha256()
+    config_qemu["hash_function"] = "sha256"
+    for fn in filenames:
+        if Path(fn).is_file():
+            hasher.update(Path(fn).read_bytes())
+        else:
+            clogger.debug("Empty file path during backup hash")
+
+    config_qemu["hash"] = hasher.digest()
+    backup = {
+        "index": -3,
+        "config": config_qemu,
+        "raw_faultlist": raw_faultlist,
+        "expanded_faultlist": expanded_faultlist,
+    }
+
+    queue_output.put(backup)
+
+
+def read_backup(hdf5_file):
+    """
+    :param hdf5_file: path to hdf5
+    :return: backup_raw_faults, backup_expanded_faults, backup_config, backup_goldenrun
+    """
+    with tables.open_file(hdf5_file, "r") as f_in:
+        # Process start points
+        startpoints = []
+        for startpoint in f_in.root.Backup.start_addresses:
+            startpoints.append(
+                {"address": startpoint["address"], "counter": startpoint["counter"]}
+            )
+
+        # Process end points
+        endpoints = []
+        for endpoint in f_in.root.Backup.end_addresses:
+            endpoints.append(
+                {"address": endpoint["address"], "counter": endpoint["counter"]}
+            )
```

Comment:
This could be performed with a list comprehension, like:
```
endpoints = [
    {"address": endpoint["address"], "counter": endpoint["counter"]} 
    for endpoint in f_in.root.Backup.end_addresses
]
```